frappe/public/js/frappe/utils/diffview.js

```diff
@@ -0,0 +1,110 @@
+frappe.provide("frappe.ui");
+
+frappe.ui.DiffView = class DiffView {
+	constructor(doctype, fieldname, docname) {
+		this.dialog = null;
+		this.handler = null;
+		this.doctype = doctype;
+		this.fieldname = fieldname;
+		this.docname = docname;
+
+		this.dialog = this.make_dialog();
+		this.dialog.show();
+	}
+
+	make_dialog() {
+		const get_query = () => ({
+			query: "frappe.utils.diff.version_query",
+			filters: { docname: this.docname, ref_doctype: this.doctype },
+		});
+		const onchange = () => this.compute_diff();
+		let dialog = new frappe.ui.Dialog({
+			title: __("Compare Versions"),
+			fields: [
+				{
+					label: __("From version"),
+					fieldtype: "Link",
+					fieldname: "from_version",
+					options: "Version",
+					reqd: 1,
+					get_query,
+					onchange,
+				},
+				{
+					fieldtype: "Column Break",
+					fieldname: "cb",
+				},
+				{
+					label: __("To version"),
+					fieldtype: "Link",
+					fieldname: "to_version",
+					options: "Version",
+					reqd: 1,
+					get_query,
+					onchange,
+				},
+				{
+					fieldtype: "Section Break",
+					fieldname: "sb",
+				},
+				{
+					label: __("Diff"),
+					fieldtype: "HTML",
+					fieldname: "diff",
+				},
+			],
+			size: "large",
+		});
+		return dialog;
+	}
+
+	compute_diff() {
+		const from_version = this.dialog.get_value("from_version");
+		const to_version = this.dialog.get_value("to_version");
+		const fieldname = this.fieldname;
+
+		if (from_version && to_version) {
+			frappe
+				.xcall("frappe.utils.diff.get_version_diff", {
+					from_version,
+					to_version,
+					fieldname,
+				})
```

Comment:
Should we blur the dom when this is getting loaded to convey the loading state to the user?

Reply:
frappe.xcall doesn't support `freeze`, also freeze doesn't freeze the dialog box just the page. 

These are just 2 simple select queries, doesn't take much time.